lib/datadog/statsd/telemetry.rb

```diff
@@ -4,29 +4,38 @@
 module Datadog
   class Statsd
     class Telemetry
-      attr_accessor :metrics
-      attr_accessor :events
-      attr_accessor :service_checks
-      attr_accessor :bytes_sent
-      attr_accessor :bytes_dropped
-      attr_accessor :packets_sent
-      attr_accessor :packets_dropped
-      attr_reader   :estimate_max_size
+      attr_reader :metrics
+      attr_reader :events
+      attr_reader :service_checks
+      attr_reader :bytes_sent
+      attr_reader :bytes_dropped
+      attr_reader :packets_sent
+      attr_reader :packets_dropped
+      attr_reader :estimate_max_size
 
-      def initialize(disabled, tags, flush_interval)
+      def initialize(disabled, flush_interval, global_tags: [], transport_type: :udp)
         @disabled = disabled
-        @tags = tags
         @flush_interval = flush_interval
+        @global_tags = global_tags
+        @transport_type = transport_type
         reset
 
+        # TODO: Karim: I don't know why but telemetry tags are serialized
+        # before global tags so by refactoring this, I am keeping the same behavior
+        @serialized_tags = Serialization::TagSerializer.new(
+          client: 'ruby',
+          client_version: VERSION,
+          client_transport: transport_type,
+        ).format(global_tags)
```

Comment:
Can `global_tags` change half-way through the execution of a program? In that case doing the `format` once here means we won't pick up new `global_tags` if they change. I'm not sure if this is a real problem, though (wdyt @remeh?)

Reply:
I think they will never change. Usually in ruby we prefer obvious behavior. Giving an array to a class and then mutating sounds really weird to me. @remeh wdyt?

Reply:
Yeah, I would also expect that, but maybe some users could be relying in it? Wouldn't be the strangest thing I've seen :P And what worries most is that it won't break their code: it would just change the behaviour silently.